### PR TITLE
rebase and structure sdk migration

### PR DESCRIPTION
**Pull request by mamy-CS**
Design Migration Doc - https://docs.google.com/document/d/1wpNMog34bhDUDeEMDrKCZhMHysQLMmsiQg3jOfACkRM/edit
Moved previous pr (https://github.com/openshift/instaslice-operator/pull/7) work here

### CODE REVIEW

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -57,7 +57,7 @@ spec:
       #             values:
       #               - linux
       securityContext:
-        runAsNonRoot: false
+        runAsNonRoot: true
```
Comment (harche) [inline config/manager/manager.yaml:60]: Does it work without this privilege?

**Comment by asm582** [comment]
All e2e local tests now pass. Thanks @mamy-CS for this PR.

/lgtm

**Pull request merged**